File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        ans=0
        temp=x
        while x > 0:
            dig=x%10
            ans=ans*10+dig
            x//=10
        if ans==temp:
            return True
        else:
            return False
        

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
   if (x < 0) return false;

        // Convert the integer to a string
        String valueString = String.valueOf(x);

        // Loop through half of the string
        for (int i = 0; i < valueString.length() / 2; i++) {
            // Compare characters from the start and end
            if (valueString.charAt(i) != valueString.charAt(valueString.length() - 1 - i)) {
                return false; // Not a palindrome
            }
        }
        return true; // It's a palindrome
    }
}